src/private/cstr.rs

```diff
@@ -19,6 +19,13 @@ pub fn set_str(buf: &mut [u8], s: &str) -> Result<(), crate::sys::EspError> {
     Ok(())
 }
 
+#[cfg(feature = "alloc")]
+pub fn set_cchar_str(buf: &mut [c_char], s: &str) -> Result<(), crate::sys::EspError> {
+    let buf_u8 = unsafe { buf.as_mut_ptr() as *mut u8 };
+    let slice = unsafe { std::slice::from_raw_parts_mut(buf_u8, buf.len()) };
+    set_str(slice, s)
```

Comment:
I don't really understand why we should reuse that function, it seems it does an allocation (in `to_cstring_arg`), which I don't think is necessary.